waspc/cli/src/Wasp/Cli/Command/CreateNewProject/Options.hs

```diff
@@ -0,0 +1,95 @@
+module Wasp.Cli.Command.CreateNewProject.Options
+  ( getProjectInfo,
+    NewProjectArgs (..),
+  )
+where
+
+import Control.Monad.IO.Class (liftIO)
+import Data.List (intercalate)
+import Options.Applicative (defaultPrefs, execParserPure)
+import qualified Options.Applicative as Opt
+import Wasp.Analyzer.Parser (isValidWaspIdentifier)
+import Wasp.Cli.Command (Command)
+import Wasp.Cli.Command.Call (Arguments)
+import Wasp.Cli.Command.CreateNewProject.Common (ProjectInfo (..), throwProjectCreationError)
+import Wasp.Cli.Command.CreateNewProject.Http (getTemplatesJson)
+import qualified Wasp.Cli.Question as Question
+import Wasp.Util (indent, kebabToCamelCase)
+
+data NewProjectArgs = NewProjectArgs
+  { _optsProjectName :: Maybe String,
+    _optsTemplateName :: Maybe String
+  }
+
+type Templates = [String]
+
+getProjectInfo :: Arguments -> Command ProjectInfo
+getProjectInfo newArgs = do
+  newProjectArgs <- getNewProjectArgs newArgs
+  templates <- liftIO getTemplatesJson
+  createProjectInfoFromArgs newProjectArgs templates
+
+createProjectInfoFromArgs :: NewProjectArgs -> Maybe Templates -> Command ProjectInfo
+createProjectInfoFromArgs (NewProjectArgs projectNameArg templateNameArg) maybeTemplates =
+  case (projectNameArg, templateNameArg, maybeTemplates) of
+    -- Non-interactive mode
+    (Just projectName, templateName, _) ->
+      createProjectInfo projectName templateName
+    -- Interactive mode
+    (_, _, Just templates) -> do
```

Comment:
Now it's called `createNewProjectDescription` and it handles the creation process based on `args` and list of `templates`. I didn't want to complicate since this is the central function that based on the available data decides either to create the new project description or interactively ask the user 🤷